tlair/src/main/scala/at/forsyte/apalache/tla/lir/io/CounterexampleWriter.scala

```diff
@@ -0,0 +1,102 @@
+package at.forsyte.apalache.tla.lir.io
+
+import java.io.PrintWriter
+
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.oper._
+import at.forsyte.apalache.tla.lir.values._
+/**
+ * A printer for counterexamples, in various formats: TLA+ , as TLC output, ...
+ *
+ * @author Andrey Kuprianov
+ */
+abstract class CounterexampleWriter {
+  // Print out invariant violation
+  def write(notInvariant: NotInvariant, init: State, nextStates: List[NextState] ) : Unit
+
+  def printState(pretty: PrettyWriter, state: State) = {
```

Comment:
This method seems to be specialized for the outputs in TLC and Apalache format. It would not be useful for JSONWriter, which we will definitely have in the future.